content/blog/open-source-monitoring-at-the-hpe-customer-technology-center-böblingen.md

```diff
@@ -0,0 +1,377 @@
+    node.labels(cn,server['Rack'],'Power','Minimum').set(server_metrics["PowerMeter"]['Minimum'])
+    for temperature in server_metrics['Temperatures']:
+        if temperature['Status']['State'] == 'Enabled': 
+            node.labels(cn,server['Rack'],'Temperature',temperature["Name"]).set(temperature['ReadingCelsius'])
+    return 0
+```
+
+The complete most current ILO Prometheus connector (iloPromConnector.v4.3.py at the time of writing this blog) together with a routine (createILOcreadentials.v1.0.py) to generate XML-input file with encrypted user credentials is available on [Github](https://github.com/tbeha/iloPrometheus).
+
+# V﻿isualize the data
+
+The CTC monitor is using Prometheus as a time series database to store and Grafana to visualize the gathered data. Prometheus, Grafana, the  [HPE Storage Array Exporter for Prometheus](https://developer.hpe.com/blog/get-started-with-prometheus-and-grafana-on-docker-with-hpe-storage-array-exporter/) and the [HPE SimpliVity Exporter](https://www.hpe.com/psnow/doc/a50000514enw) are deployed as Kubernetes pods in order to easily scale and adjust the CTC monitoring environment. 
+
+In order to include the CTC power and temperature monitoring into this environment I first needed to containerize the HPE Server Exporter.  I based the container on Ubuntu Linux, added Python and the necessary Python libraries to it and stored the server exporter script in the directory /opt/prometheus:
+
+```dockerfile
+# User Ubuntu as the base Image
+FROM ubuntu
+#
+LABEL maintainer="Thomas Beha"
+LABEL version="4.3"
+LABEL copyright="Thomas Beha, 2023"
+LABEL license="GNU General Public License v3"
+LABEL DESCRIPTION="CTC ILO Redfish Prometheus Connector Python container based on Ubuntu"
+#
+RUN apt-get update
+RUN apt-get -y install python3.6 && \
+	apt-get -y install python3-pip  && \
+	apt-get -y install iputils-ping
+RUN /usr/bin/pip3 install requests && \
+	/usr/bin/pip3 install fernet && \
+	/usr/bin/pip3 install cryptography && \
+	/usr/bin/pip3 install lxml && \
+	/usr/bin/pip3 install redfish && \
+	/usr/bin/pip3 install prometheus_client
+# copy the necessary python files to the container
+RUN mkdir /opt/prometheus
+COPY ./iloPromConnector.v4.3.py /opt/prometheus
+```
+
+I used this Dockerfile to generate a container image and upload it into my registry. Before I could define the Kubernetes pod based on this container image, I had to decide how to provide the input data to the pod. The current HPE server exporter expects to have the input data available at /opt/prometheus/data. The easiest and most flexible way to provide the input data was to use a Kubernetes config map. This allowed me to use the same container image multiple times, each time with a different config map. The large number of server hosted in the CTC required multiple server exporter, each one with a different list of server to monitor. An example of a config map, that I used is given below:
```

Comment:
Change "upload" to "uploaded" in the first sentence.

I'm not sure on this one, but I think the 4th sentence should actually be in the present tense, i.e.:
"The easiest and most flexible way to provide the input data is to use a Kubernetes config map." because no matter when you did it, this will always still be the case, right?

"The large number of server" should read "The large number of servers" (plural) - which means that the Multiple server exporter should also be "exporters". Later in that sentence, it should read "with a different list of servers to monitor." (again, plural)

Change the last sentence to read "An example of a config map that I used is shown below:" (no comma, and change "given" to "shown")